--- a/library/std/src/sys/pal/unix/sync/condvar.rs
+++ b/library/std/src/sys/pal/unix/sync/condvar.rs
@@ -67,7 +67,7 @@
         #[cfg(target_os = "cygwin")]
         let dur = Duration::min(dur, Duration::from_secs(1000 * 365 * 86400));
 
-        let timeout = Timespec::now(Self::CLOCK).checked_add_duration(&dur);
+        let timeout = Timespec::now(Self::CLOCK()).checked_add_duration(&dur);
 
         #[cfg(not(target_os = "nto"))]
         let timeout = timeout.and_then(|t| t.to_timespec()).unwrap_or(TIMESPEC_MAX);
@@ -145,7 +145,11 @@
 )))]
 impl Condvar {
     pub const PRECISE_TIMEOUT: bool = true;
-    const CLOCK: libc::clockid_t = libc::CLOCK_MONOTONIC;
+    // cosmo patch: libc::CLOCK_MONOTONIC is extern static.
+    #[allow(non_snake_case)]
+    fn CLOCK() -> libc::clockid_t {
+        unsafe { libc::CLOCK_MONOTONIC }
+    }
 
     /// # Safety
     /// May only be called once per instance of `Self`.
@@ -167,7 +171,7 @@
             let r = libc::pthread_condattr_init(attr.as_mut_ptr());
             assert_eq!(r, 0);
             let attr = AttrGuard(&mut attr);
-            let r = libc::pthread_condattr_setclock(attr.0.as_mut_ptr(), Self::CLOCK);
+            let r = libc::pthread_condattr_setclock(attr.0.as_mut_ptr(), Self::CLOCK());
             assert_eq!(r, 0);
             let r = libc::pthread_cond_init(self.raw(), attr.0.as_ptr());
             assert_eq!(r, 0);
